### DISCUSSION TITLE
Code not printing values in trained_epoch_end

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Lightning-AI/pytorch-lightning#11851] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
My code is not printing `print('train acc loss',acc,loss)` in `trained_epoch_end` but its printing `print('val acc loss',acc,loss)` in `validation_epoch_end` 
```
class Model(LightningModule):
  def __init__(self):
    super(Model,self).__init__()
    self.model=ResneT(21)
    self.lr=1e-3
    self.bs=128
    self.worker=6
    self.acc=torchmetrics.Accuracy()
    self.creterion=nn.BCEWithLogitsLoss()
    self.scheduler='lambda'
  def forward(self,x):
    x=self.model(x)
    return x

  def configure_optimizers(self):
    opt=torch.optim.AdamW(params=self.parameters(),lr=self.lr )
    return opt

  def train_dataloader(self):
    dataset=DataReader(train_df)
    dataloader=DataLoader(dataset,batch_size=self.bs,num_workers=self.worker,shuffle=True,
                         pin_memory=True,collate_fn=collate_fn)
    return dataloader

  def training_step(self,batch,batch_idx):
    signal,label=batch
    out=self(signal.float())
    loss=self.creterion(out.flatten(),label.float().flatten())
    acc=self.acc(out.flatten(),label.long().flatten())
    return {'loss':loss,'acc':acc}

  def trained_epoch_end(self,outputs):
    acc=torch.stack([x['acc'] for x in outputs]).mean().detach().cpu().numpy().round(2)
    loss=torch.stack([x['loss'] for x in outputs]).mean().detach().cpu().numpy().round(2)
    self.log('train acc',acc)
    self.log('train loss',loss)
    print('train acc loss',acc,loss)

  def val_dataloader(self):
    dataset=DataReader(val_df)
    dataloader=DataLoader(dataset,batch_size=self.bs,num_workers=self.worker,shuffle=False,
                          pin_memory=True,
                         collate_fn=collate_fn)
    return dataloader

  def validation_step(self,batch,batch_idx):
    signal,label=batch
    out=self(signal.float())
    loss=self.creterion(out.flatten(),label.float().flatten())
    acc=self.acc(out.flatten(),label.long().flatten())
    return {'loss':loss,'acc':acc}

  def validation_epoch_end(self,outputs):
    acc=torch.stack([x['acc'] for x in outputs]).mean().detach().cpu().numpy().round(2)
    loss=torch.stack([x['loss'] for x in outputs]).mean().detach().cpu().numpy().round(2)
    print('val acc loss',self.current_epoch,acc,loss)
    self.log('val acc',acc)
    self.log('val loss',loss)
```

**Comment by ananthsub** [comment, marked as answer]
You need to implement `training_epoch_end`, not `trained_epoch_end`: https://github.com/PyTorchLightning/pytorch-lightning/blob/1515ef90ee2724bcba46e1434eb4b4f9719ebdd7/pytorch_lightning/core/lightning.py#L689

**Reply by talhaanwarch** [reply]
thanks. it resolved
